### PR TITLE
fix: storage conflict errors return 409 instead of 500

### PR DESCRIPTION
**Pull request by adriantam**
## Description
In the case of error is write failure is due to ErrTransactionalWriteFailed.  We should return 409 instead of 500 error.

## References
Close https://github.com/openfga/openfga/issues/1080
Corresponding change https://github.com/openfga/api/pull/107

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by jon-whit** [comment]
@adriantam please cross-reference with the update to the protobuf API for the OpenAPI changes.

**Review by jon-whit — DISMISSED** [review]

File: pkg/server/commands/write.go [hunk 1]
```diff
@@ -137,7 +139,7 @@ func (c *WriteCommand) validateNoDuplicatesAndCorrectSize(deletes []*openfgav1.T
 
 func handleError(err error) error {
 	if errors.Is(err, storage.ErrTransactionalWriteFailed) {
```
Comment (miparnisari) [inline pkg/server/commands/write.go:141]: can you update the text of this error, too?

https://github.com/openfga/openfga/blob/aa0c8162003a389263ee27c6be6731773bda0147/pkg/storage/errors.go#L18

to "transactional write failed due to conflict"

File: pkg/server/commands/write.go [hunk 2]
```diff
@@ -137,7 +139,7 @@ func (c *WriteCommand) validateNoDuplicatesAndCorrectSize(deletes []*openfgav1.T
...
-		return serverErrors.NewInternalError("concurrent write conflict", err)
+		return status.Error(codes.Aborted, "transaction aborted")
```
Comment (miparnisari) [inline pkg/server/commands/write.go:142]: should this be `status.Error(codes.Aborted, err)`? i'd like not use the text coming from the error itself

File: pkg/server/errors/encoded_errors.go [hunk 3]
```diff
@@ -68,6 +68,18 @@ func sanitizedMessage(message string) string {
 // NewEncodedError returns the encoded error with the correct http status code etc.
 func NewEncodedError(errorCode int32, message string) *EncodedError {
 	if !IsValidEncodedError(errorCode) {
+		// Check to see if it is aborted
+		if errorCode == int32(codes.Aborted) {
```
Comment (miparnisari) [inline pkg/server/errors/encoded_errors.go:72]:
```suggestion
		if errorCode == int32(codes.Aborted) {
```

File: pkg/server/commands/write_test.go [hunk 4]
```diff
@@ -229,14 +231,8 @@ type document
...
-	require.ErrorIs(
-		t,
-		err,
-		serverErrors.NewInternalError(
-			"concurrent write conflict",
-			storage.ErrTransactionalWriteFailed,
-		),
-	)
+
+	require.ErrorIs(t, err, status.Error(codes.Aborted, "transaction aborted"))
```
Comment (miparnisari) [inline pkg/server/commands/write_test.go:235]:
```suggestion
	require.ErrorIs(t, err, status.Error(codes.Aborted, storage.ErrTransactionalWriteFailed.Error()))
```

**Review by miparnisari — APPROVED** [review]

**Pull request merged**